components/Account/NewsletterSubscriptions.js

```diff
@@ -80,7 +98,59 @@ const NewsletterSubscriptions = props => (
         <Fragment>
           {status !== 'subscribed' && (
             <Box style={{ margin: '10px 0', padding: 15 }}>
-              <P>{t('account/newsletterSubscriptions/unsubscribed')}</P>
+              <Mutation mutation={RESUBSCRIBE_EMAIL}>
+                {(mutate, { loading, error, data: mutationData }) => (
+                  <>
+                    {status === 'unsubscribed' && (
+                      <P>{t('account/newsletterSubscriptions/unsubscribed')}</P>
+                    )}
+                    {/* Show if the status has been set to pending */}
+                    {!error &&
+                      mutationData?.resubscribeEmail?.status === 'pending' && (
+                        <P>
+                          {t('account/newsletterSubscriptions/resubscribed')}
+                        </P>
+                      )}
+                    {/* Show if the status is pending an no new email has been requested */}
+                    {status === 'pending' && !mutationData && (
+                      <P>
+                        {t(
+                          'account/newsletterSubscriptions/resubscribeEmailPending'
+                        )}
+                      </P>
+                    )}
+                    <div style={{ marginTop: 10 }}>
+                      {!mutationData && (
+                        <>
+                          {error && <ErrorMessage error={error} />}
```

Comment:
As far as I can see this could be moved up and simplified:
```suggestion
                    {!mutationData && <div style={{ marginTop: 10 }}>
                      {error && <ErrorMessage error={error} />}
```
(also needs adjustment at the end, changes: rm fragment and move condition around div)